### PR TITLE
feat/FOROME-1173" create pagination list

### PR DESCRIPTION
**Pull request by QSdmitrioul**
[Task](https://quantori.atlassian.net/browse/FOROME-1173)
[Another Task](https://quantori.atlassian.net/browse/FOROME-1183)
Also move folder with shodow scroller to components folder.

### CODE REVIEW

**Review by Lapk1n — APPROVED** [review]

**Review by Lapk1n — CHANGES_REQUESTED** [review]
1. Open FR for xl_PGP3140_wgs_NIST-4_2
2. Select Panels attribute
3. Go to the second page
4. Type "P" in the search field

File: src/components/pagination-list/pagination-list.hook.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+import { PageElements } from '@components/pagination-list/pagination-list.interfaces'
+import {
+  findIndexOfLastElement,
+  getBottomPosition,
+  getDefaultCount,
+} from '@components/pagination-list/pagination-list.utils'
+
+interface IPaginationListState {
+  page: number
+  pageElements: PageElements[]
```
Comment (AndreyUstyumenko) [inline src/components/pagination-list/pagination-list.hook.ts]: PageElement

File: src/components/pagination-list/pagination-list.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import { PropsWithChildren, ReactNode, useRef } from 'react'
+import cn, { Argument } from 'classnames'
+
+import { t } from '@i18n'
+import { PaginationFooter } from '@components/pagination-list/components/pagination-footer'
+import { usePagination } from '@components/pagination-list/pagination-list.hook'
+
+interface IPaginationListProps<T> {
+  defaultCount?: number
```
Comment (AndreyUstyumenko) [inline src/components/pagination-list/pagination-list.tsx]: Do you use this 'defaultCount' somehow? And 'defaultCount' is not an obious name, maybe better 'defaultItemsCount'?

**Comment by AndreyUstyumenko** [comment]
Go to any XL dataset, open FR and choose'Symbol' attribute and start filtering results by using search. For example you can type in 'x' or 'z'. Results will be filtered. Then clear the search bar. Then change the page forwards multiple times, and try to filter by for the same symbol, you'll get different results. If you try to type in more symbols (for example 'zz'), the whole application freezes and stops responding.

**Comment by AndreyUstyumenko** [comment]
I think we need to graphically simplify the component. 'Prev' and 'Next' should be replaced with < and > icons and the phrase 'From ... to ...' should look like 1 -- 25 / 19726
Screen Shot 2022-06-13 at 15 15 51

UPD: I created a task for Anastasia to re-design the component https://quantori.atlassian.net/browse/FOROME-1222

**Comment by AndreyUstyumenko** [comment]
UPD: Decided to go back to the previous design.

File: src/components/pagination-list/pagination-list.hook.ts (hunk 1 above)
Comment (QSdmitrioul) [reply src/components/pagination-list/pagination-list.hook.ts]: Fixed

File: src/components/pagination-list/pagination-list.tsx (hunk 2 above)
Comment (QSdmitrioul) [reply src/components/pagination-list/pagination-list.tsx]: Delete it

**Review by AndreyUstyumenko — APPROVED** [review]

**Review by Lapk1n — APPROVED** [review]

**Pull request merged**